### PR TITLE
User can view collection of products

### PR DESCRIPTION
**Pull request by miyeaier**
## User Story
```
As a visitor,
In order to get an overview of the products
I would like to be able to see a catalog
```
Screenshot 2022-05-04 at 16 09 25

Screenshot 2022-05-04 at 16 09 48

Screenshot 2022-05-04 at 11 52 57

```
PS: IF YOU CAN'T FIND "Create a menu component" Shagufta did on breach  header so everything is done for  User can view collection of products
```

### CODE REVIEW

File: .nyc_output/out.json [hunk 1]
```diff
@@ -94,6 +319,6 @@
     "f": {},
     "b": {},
     "_coverageSchema": "1a1c01bbd47fc00a2c39e90264f33305004495a9",
-    "hash": "7c8e731bd4cc34f700a6cad6ff919c39f9cdbffb"
+    "hash": "deaffac37b29f0c57fa7e42cf2ce61997b0700d0"
```
Comment (tochman) [inline .nyc_output/out.json]: This entire file needs to be removed from version control

File: cypress/fixtures/Products.json [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    {
+      "id": 3,
+      "name":"Jersey Shirt",
+      "category":"Shirt",
+      "image": "https://images.pexels.com/photos/159555/soccer-football-athlete-player-159555.jpeg?auto=compress&cs=tinysrgb&w=1260&h=750&dpr=2",
+      "price": "100kr",
+      "description": " Red Nike Jersey Shirt "
+    }
+  ]
+}
```
Comment (tochman) [inline cypress/fixtures/Products.json:27]: Please use lower case letters in file names, unless it its a component

File: cypress/integration/UserCanViewProducts.feacture.js [hunk 3]
```diff
@@ -0,0 +1,10 @@
+beforeEach(() => {
+  cy.intercept("GET", "**/api/products", {
+    fixture: "products.json",
+  }).as("getProducts");
+  cy.visit("/");
+});
+
+it("is expected to display a list with 3 items", () => {
+  cy.get("[data-cy=ProductList]").children().should("have.length", 3);
```
Comment (tochman) [inline cypress/integration/UserCanViewProducts.feacture.js]: please use `snake_case` or `kebab-case` in data attributes. In thes case `product_list` or `product-list`

File: cypress/integration/userCanSeeAnApplication.feature.js [hunk 4]
```diff
@@ -1,9 +1,12 @@
-describe('user can see an app', () => {
+describe("interface", () => {
   beforeEach(() => {
-    cy.visit('/', {})
-  })
+    cy.intercept("GET", "**/api/products", {
+      fixture: "products.json",
+    }).as("getProducts");
```
Comment (tochman) [inline cypress/integration/userCanSeeAnApplication.feature.js]: There is no need to store the intercept in an variable. Please remove that.

Comment (tochman) [reply cypress/integration/userCanSeeAnApplication.feature.js]: By the way, is this test still passing?

File: src/App.jsx [hunk 5]
```diff
@@ -1,9 +1,12 @@
...
-
+import React from "react";
+import Products from "./components/Products";
 const App = () => {
+
   return (
-    <div data-cy="name">Spoko</div>
-  )
-}
+    <div data-cy="products-list">
```
Comment (tochman) [inline src/App.jsx]: check your test file. This `cy` attribute is NOT what you are looking for in your test, or it rather, it is not used at all. Please see the `cy` attribute in `Products.jsx`

File: src/components/Products.jsx [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+            />
+            {product.description}
+          </div>
+        );
+      }
+    );
+
+
+return (
+  <div data-cy="ProductList">{productList}</div>
```
Comment (tochman) [inline src/components/Products.jsx]: See comment in test file regarding the `cy` attribute.

**Review by tochman — CHANGES_REQUESTED** [review]
See comments. Also, I can't see that all tests are passing. Can't see that "Spoko"  is being displayed. But perhaps I'm missing something.

File: cypress/fixtures/Products.json (hunk 2 above)
Comment (miyeaier) [reply cypress/fixtures/Products.json:27]: changed

Comment (miyeaier) [reply cypress/fixtures/Products.json:27]: > Please use lower case letters in file names, unless it its a component

changed

File: cypress/integration/UserCanViewProducts.feacture.js (hunk 3 above)
Comment (miyeaier) [reply cypress/integration/UserCanViewProducts.feacture.js]: > please use `snake_case` or `kebab-case` in data attributes. In thes case `product_list` or `product-list`

changed to kebab-case

File: cypress/integration/userCanSeeAnApplication.feature.js (hunk 4 above)
Comment (miyeaier) [reply cypress/integration/userCanSeeAnApplication.feature.js]: yes

Comment (miyeaier) [reply cypress/integration/userCanSeeAnApplication.feature.js]: > There is no need to store the intercept in an variable. Please remove that.

changed

File: src/App.jsx (hunk 5 above)
Comment (miyeaier) [reply src/App.jsx]: > check your test file. This `cy` attribute is NOT what you are looking for in your test, or it rather, it is not used at all. Please see the `cy` attribute in `Products.jsx`

it works

File: src/components/Products.jsx (hunk 6 above)
Comment (miyeaier) [reply src/components/Products.jsx]: > See comment in test file regarding the `cy` attribute.

no problem now

**Comment by miyeaier** [comment]
> See comments. Also, I can't see that all tests are passing. Can't see that "Spoko" is being displayed. But perhaps I'm missing something.

File: .nyc_output/out.json (hunk 1 above)
Comment (miyeaier) [reply .nyc_output/out.json]: > This entire file needs to be removed from version control

removed now

File: cypress/integration/userCanViewProducts.feacture.js [hunk 7]
```diff
@@ -0,0 +1,20 @@
+beforeEach(() => {
+  cy.intercept("GET", "**/api/products", {
+    fixture: "products.json",
+  }).as("getProducts");
```
Comment (tochman) [inline cypress/integration/userCanViewProducts.feacture.js:4]: You are still saving the intercept but not using it in tests.

Comment (miyeaier) [reply cypress/integration/userCanViewProducts.feacture.js:4]: fixed ,used now.

**Pull request merged**